remoto/connection.py

```diff
@@ -21,9 +21,16 @@ def __init__(self, hostname, logger=None, sudo=False, threads=1, eager=True,
         self.interpreter = interpreter or 'python%s' % sys.version_info[0]
 
         if eager:
-            if detect_sudo:
-                self.sudo = self._detect_sudo()
-            self.gateway = self._make_gateway(hostname)
+            try:
+                if detect_sudo:
+                    self.sudo = self._detect_sudo()
+                self.gateway = self._make_gateway(hostname)
+            except OSError:
```

Comment:
Oh but then it prevents the `exc = _execnet or execnet` trick. Really not sure at this point. I'd keep the pull request as is.